src/main/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRent.kt

```diff
@@ -0,0 +1,40 @@
+package interactor
+import model.CityEntity
+import utils.isNotNull
+import java.lang.IllegalStateException
+import java.util.NoSuchElementException
+
+class GetCityHasTheHighestDifferentInApartmentRent (
+    private val dataSource: CostOfLivingDataSource,
+)
+{
+
+    fun execute(limit: Int): CityEntity {
+        return if (limit==1)
+            dataSource.getAllCitiesData().also { if (it.isEmpty()) throw IllegalStateException("no data") }
+                .filter(::excludeNullApartmentBedroom)
+                .let { it.ifEmpty { throw NoSuchElementException("no filtered data") } }
+                .sortedByDescending { it.findHighestDifferentInCitiesRent() }
+                .take(limit)
+                .first()
+        else throw InvalidLimitException("invalid limit value")
+    }
```

Comment:
If we should return the highest city(JUST **ONE** city), why we are using limit?
```suggestion
    fun execute(): CityEntity =
        dataSource.getAllCitiesData()
            .ifEmpty { throw IllegalStateException("Something went wrong") }
            .filter(::excludeNullApartmentBedroom)
            .ifEmpty { throw NoSuchElementException("No data matching the condition.") }
            .maxByOrNull { it.findHighestDifferentInCitiesRent() }
            ?: throw NoSuchElementException("")
```
You have to change related the test cases

Reply:
add asSequence() function